chatsky/core/message.py

```diff
@@ -388,9 +390,23 @@ def __init__(  # this allows initializing Message with string as positional argu
             annotations=annotations,
             misc=misc,
             origin=origin,
+            timestamp=timestamp,
             **kwargs,
         )
 
+    def __eq__(self, other):
+        if not isinstance(other, Message):
+            return NotImplementedError
```

Comment:
This line is not covered.
`__eq__` should return `NotImplemented` instead of `NotImplementedError`.

Add a test to cover this.